flink-core/src/main/java/org/apache/flink/api/common/operators/ResourceSpec.java

```diff
@@ -145,13 +148,15 @@ private ResourceSpec() {
 	 * @param other Reference to resource to merge in.
 	 * @return The new resource with merged values.
 	 */
-	public ResourceSpec merge(ResourceSpec other) {
+	public ResourceSpec merge(final ResourceSpec other) {
+		checkNotNull(other, "Cannot merge with null resources");
+
 		if (this.equals(UNKNOWN) || other.equals(UNKNOWN)) {
 			return UNKNOWN;
 		}
 
 		ResourceSpec target = new ResourceSpec(
-				this.cpuCores + other.cpuCores,
+				this.cpuCores.merge(other.cpuCores).getValue(),
```

Comment:
So we lose the precision info here?

Reply:
BTW, it seems that the precision will always set to the default, why not change it to a constant.

Reply:
>> So we lose the precision info here?

I think not. The precision is lost only when we do calculation with it.

Reply:
>> BTW, it seems that the precision will always set to the default, why not change it to a constant.

Different precisions are support to support different extended resources in the future. Since there is no contract of the precision, each resource can decide it by themselves if needed.

Reply:
So, why the constructor of ResourceSpec just accept a `double` parameter?

Reply:
It's for backward compatibility. But I think you are right we can change the constructor to directly accept ResourceValue. The change is made in 9627a892cd7076abf01174376dbcdfc9baf41c13.